analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -0,0 +1,85 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn
+{
+    public partial class RoslynSymbolicExecutionTest
+    {
+        [TestMethod]
+        public void Invocation_IsNullOrEmpty_ValidateOrder()
+        {
+            var validator = SETestContext.CreateCS(@"var isNullOrEmpy = string.IsNullOrEmpty(arg);", ", string arg").Validator;
+            validator.ValidateOrder(
+"LocalReference: isNullOrEmpy = string.IsNullOrEmpty(arg) (Implicit)",
+"ParameterReference: arg",
+"Argument: arg",
+"Invocation: string.IsNullOrEmpty(arg)", // True/Null
+"Invocation: string.IsNullOrEmpty(arg)", // True/NotNull
+"Invocation: string.IsNullOrEmpty(arg)", // False/NotNull
+"SimpleAssignment: isNullOrEmpy = string.IsNullOrEmpty(arg) (Implicit)",  // True/Null
+"SimpleAssignment: isNullOrEmpy = string.IsNullOrEmpty(arg) (Implicit)",  // True/NotNull
+"SimpleAssignment: isNullOrEmpy = string.IsNullOrEmpty(arg) (Implicit)"); // False/NotNull
+        }
+
+        [TestMethod]
+        public void Invocation_IsNullOrEmpty_Tags()
+        {
+            const string code = @"
+var isNullOrEmpy = string.IsNullOrEmpty(arg);
+Tag(""IsNullOrEmpy"", isNullOrEmpy);
+Tag(""Arg"", arg);";
+            var validator = SETestContext.CreateCS(code, ", string arg").Validator;
+            validator.TagValues("IsNullOrEmpy").Should().Equal(
+                new SymbolicValue().WithConstraint(BoolConstraint.True),       // True/Null
+                new SymbolicValue().WithConstraint(BoolConstraint.True),       // True/NotNull
+                new SymbolicValue().WithConstraint(BoolConstraint.False));     // False/NotNull
+            validator.TagValues("Arg").Should().Equal(
+                new SymbolicValue().WithConstraint(ObjectConstraint.Null),     // True/Null
+                new SymbolicValue().WithConstraint(ObjectConstraint.NotNull),  // True/NotNull
+                new SymbolicValue().WithConstraint(ObjectConstraint.NotNull)); // False/NotNull
+        }
+
+        [TestMethod]
+        public void Invocation_IsNullOrEmpty_NestedProperty()
+        {
+            const string code = @"
+if (!string.IsNullOrEmpty(exception?.Message))
+{
+    Tag(""ExceptionChecked"", exception);
+}
+Tag(""ExceptionAfterCheck"", exception);";
+            var validator = SETestContext.CreateCS(code, ", InvalidOperationException exception").Validator;
+            validator.TagValues("ExceptionChecked").Should().Equal(new[]
+            {
+                new SymbolicValue().WithConstraint(ObjectConstraint.Null),
+                new SymbolicValue().WithConstraint(ObjectConstraint.NotNull)
+            }); // Should be 'new SymbolicValue().WithConstraint(ObjectConstraint.NotNull)'
```

Comment:
It's not clear to me what assertion this line refers to. Let's add `// This should not be here` to the relevant line (s)